File updates

JustCat80 opened the pull request:
Recreated from #1273

Deprecates file for files

Also adds support for editing attachments within a message with editMessage

Commit:
File changes for createMessage and editMessage

Changes createMessage and editMessage

Changes to createMessage

Change createMessage to use content.file, edit docs, also edit docs to show that you can send an array of files in channels

Add content.file to editMessage

Add content.file to editMessage and jsdocs

Add ts docs

Oops forgot Message.edit jsdocs

Make tsdocs deprecation consistent with the rest

File checks

Add file check to editMessage
Fix check in createMessage

Commit:
Merge branch 'dev' of https://github.com/abalabahaha/eris into file-updates

Commit:
Move deprecated file check so that it cannot error if only a string is provided

JustCat80 committed:
content.attachments rather then content.file

JustCat80 committed:
Update jsdocs, change wording

bsian03 commented on lib/Client.js:
```diff
@@ -609,7 +609,10 @@ class Client extends EventEmitter {
+    * @arg {Array<Object>} [content.attachments] An array of attachment objects
+    * @arg {Buffer} content.attachments.file A buffer containing attachment data
+    * @arg {String} content.attachments.name What to name the attachment
+    * @arg {Object | Array<Object>} [file] [DEPRECATED] A file object (or an Array of them). Use `content.attachments` instead
```
Alphabetical placement

bsian03 commented on lib/Client.js:
```diff
@@ -1248,6 +1257,9 @@ class Client extends EventEmitter {
     * @arg {Boolean} [content.allowedMentions.everyone] Whether or not to allow @everyone/@here.
     * @arg {Boolean | Array<String>} [content.allowedMentions.roles] Whether or not to allow all role mentions, or an array of specific role mentions to allow.
     * @arg {Boolean | Array<String>} [content.allowedMentions.users] Whether or not to allow all user mentions, or an array of specific user mentions to allow.
+    * @arg {Object | Array<Object>} [content.attachments] An array of attachment objects
```
Object ain't an array

bsian03 commented on lib/Client.js:
```diff
@@ -1539,7 +1551,10 @@ class Client extends EventEmitter {
     * @arg {String} [options.content=""] A content string
     * @arg {Array<Object>} [options.embeds] An array of Discord embeds
-    * @arg {Object | Array<Object>} [options.file] A file object (or an Array of them)
+    * @arg {Object | Array<Object>} [options.attachments] An array of attachment objects
```
Object ain't an array, also alphabetical ordering

bsian03 commented on lib/Client.js:
```diff
@@ -1539,7 +1551,10 @@ class Client extends EventEmitter {
+    * @arg {String} options.attachments.name What to name the attachment
+    * @arg {Object | Array<Object>} [options.file] [DEPRECATED] A file object (or an Array of them), use options.attachments instead
     * @arg {Buffer} options.file.file A buffer containing file data
     * @arg {String} options.file.name What to name the file
```
Could probably just get rid of these 2

bsian03 commented on lib/Client.js:
```diff
@@ -1615,7 +1636,10 @@ class Client extends EventEmitter {
     * @arg {String} [options.content=""] A content string
     * @arg {Array<Object>} [options.embeds] An array of Discord embeds
-    * @arg {Object | Array<Object>} [options.file] A file object (or an Array of them)
+    * @arg {Object | Array<Object>} [options.attachments] An array of attachment objects
```
Object ain't an array, also alphabetical ordering

bsian03 commented on lib/Client.js:
```diff
@@ -1615,7 +1636,10 @@ class Client extends EventEmitter {
+    * @arg {String} options.attachments.name What to name the attachment
+    * @arg {Object | Array<Object>} [options.file] [DEPRECATED] A file object (or an Array of them), use options.attachments instead
     * @arg {Buffer} options.file.file A buffer containing file data
     * @arg {String} options.file.name What to name the file
```
Could just get rid of these 2

bsian03 commented on lib/Client.js:
```diff
@@ -1632,7 +1656,7 @@ class Client extends EventEmitter {
             allowed_mentions: this._formatAllowedMentions(options.allowedMentions),
             components: options.components
-        }, options.file).then((response) => options.wait ? new Message(response, this) : undefined);
+        }, options.attachments).then((response) => options.wait ? new Message(response, this) : undefined);
```
Breaking deprecation compatibility

bsian03 reviewed (changes requested):
Please make relevant changes on individual structure classes too

JustCat80 committed:
Add requested changes

Fix jsdocs
Alphabetize
Fix deprecation

JustCat80 committed:
Merge branch 'dev' of https://github.com/abalabahaha/eris into file-updates

bsian03 commented on lib/Client.js:
```diff
@@ -1588,16 +1603,20 @@ class Client extends EventEmitter {
-    * @arg {Object | Array<Object>} [options.file] A file object (or an Array of them)
-    * @arg {Buffer} options.file.file A buffer containing file data
-    * @arg {String} options.file.name What to name the file
+    * @arg {Object | Array<Object>} [options.file] [DEPRECATED] A file object (or an Array of them), use options.attachments instead
```
Inconsistent deprecation message with createMessage (and other locations)

JustCat80 replied:
fixed

bsian03 commented on lib/Client.js:
```diff
@@ -1664,15 +1686,19 @@ class Client extends EventEmitter {
-    * @arg {Object | Array<Object>} [options.file] A file object (or an Array of them)
-    * @arg {Buffer} options.file.file A buffer containing file data
-    * @arg {String} options.file.name What to name the file
+    * @arg {Object | Array<Object>} [options.file] [DEPRECATED] A file object (or an Array of them), use options.attachments instead
```
Same here (ignore this if this gets included in the review, i tried to delete it and couldn't for some reason)

JustCat80 committed:
Consistent file deprecation warnings

JustCat80 committed:
Change attachments to its own argument

as content.attachments is used in some places
document content.attachments
Fix jsdocs

JustCat80 committed:
Oops fix jsdocs

DonovanDMC commented on index.d.ts:
```diff
@@ -845,6 +847,7 @@ declare namespace Eris {
   }
   interface AdvancedMessageContent {
     allowedMentions?: AllowedMentions;
+    attachments?: {id: string, filename: string};
```
this should be made into its own interface or separated out into different lines, it looks out of place as-is

that should also be a semicolon, not a comma
```suggestion
    attachments?: {
        id: string;
        filename: string
    };
```

JustCat80 replied:
I didn't realize Attachment was already an interface lol

JustCat80 committed:
Update content.attachments

AdvancedMessageContent.attachments is an array of Attachment

JustCat80 committed:
attachments => files

JustCat80 committed:
Merge branch 'dev' of https://github.com/abalabahaha/eris into file-updates

JustCat80 committed:
Merge branch 'dev' of https://github.com/abalabahaha/eris into file-updates

JustCat80 committed:
Change interaction things from file to files

JustCat80 committed:
Files => content/options.files

JustCat80 committed:
Merge branch 'dev' of https://github.com/abalabahaha/eris into file-updates

JustCat80 committed:
Fix some jsdoc mistakes

JustCat80 committed:
Merge pull request #75 from ErisPRUpdateBot/upstream/catboi8/file-updates

Upstream Update (file-updates)
